src/games/strategy/triplea/ai/proAI/util/ProMoveOptionsUtils.java

```diff
@@ -1163,88 +1144,94 @@ private void findBombardOptions(final PlayerID player, final List<Territory> myU
+          if (!data.getMap().getNeighbors(capital).contains(t)) {
+
+            // Get max enemy defenders
+            final Set<Unit> additionalEnemyDefenders = new HashSet<Unit>();
+            final List<PlayerID> players = ProUtils.getOtherPlayersInTurnOrder(player);
+            for (final ProTerritory enemyDefendOption : enemyDefendOptions.getAll(t)) {
+              final Set<Unit> enemyUnits = new HashSet<Unit>(enemyDefendOption.getMaxUnits());
+              enemyUnits.addAll(enemyDefendOption.getMaxAmphibUnits());
+              if (!enemyUnits.isEmpty()) {
+                final PlayerID enemyPlayer = enemyUnits.iterator().next().getOwner();
+                if (ProUtils.isPlayersTurnFirst(players, enemyPlayer, alliedPlayer)) {
+                  additionalEnemyDefenders.addAll(enemyUnits);
+                }
               }
             }
-          }
 
-          // Check allied result without strafe
-          final Set<Unit> enemyDefendersBeforeStrafe = new HashSet<Unit>(defenders);
-          enemyDefendersBeforeStrafe.addAll(additionalEnemyDefenders);
-          final ProBattleResult result =
-              battleUtils.estimateAttackBattleResults(alliedPlayer, t, new ArrayList<Unit>(alliedUnits),
-                  new ArrayList<Unit>(enemyDefendersBeforeStrafe), alliedAttack.getMaxBombardUnits());
-          if (result.getWinPercentage() < WIN_PERCENTAGE) {
-            patd.setStrafing(true);
-
-            // Try to strafe to allow allies to conquer territory
-            final Set<Unit> combinedUnits = new HashSet<Unit>(patd.getMaxUnits());
-            combinedUnits.addAll(patd.getMaxAmphibUnits());
-            final ProBattleResult strafeResult =
-                battleUtils.callBattleCalculator(player, t, new ArrayList<Unit>(combinedUnits), defenders,
-                    patd.getMaxBombardUnits(), true, true);
-
-            // Check allied result with strafe
-            final Set<Unit> enemyDefendersAfterStrafe = new HashSet<Unit>(strafeResult.getAverageDefendersRemaining());
-            enemyDefendersAfterStrafe.addAll(additionalEnemyDefenders);
-            patd.setMaxBattleResult(battleUtils.estimateAttackBattleResults(alliedPlayer, t, new ArrayList<Unit>(
-                alliedUnits), new ArrayList<Unit>(enemyDefendersAfterStrafe), alliedAttack.getMaxBombardUnits()));
-
-
-            ProLogger.debug("Checking strafing territory: " + t + ", alliedPlayer="
-                + alliedUnits.iterator().next().getOwner().getName() + ", maxWin%="
-                + patd.getMaxBattleResult().getWinPercentage() + ", maxAttackers=" + alliedUnits.size()
-                + ", maxDefenders=" + enemyDefendersAfterStrafe.size());
-
-            if (patd.getMaxBattleResult().getWinPercentage() >= WIN_PERCENTAGE) {
-              System.out.println(data.getSequence().getRound() + ". strafing territory: " + t + ", alliedPlayer="
+            // Check allied result without strafe
```

Comment:
long method code smell. Functionality could get broken up into private methods, and then perhaps moved into a helper object if there is a lot of common data. In both cases it allows for testing since the 'private' methods could be promoted to protected for testing, and/or the helper object could be testd.